tqdm/contrib/__init__.py

```diff
@@ -15,10 +15,18 @@
 
 class DummyTqdmFile(ObjectWrapper):
     """Dummy file-like that will write to tqdm"""
+    def __init__(self, wrapped):
+        super(DummyTqdmFile, self).__init__(wrapped)
+        self._buf = []
+
     def write(self, x, nolock=False):
-        # Avoid print() second call (useless \n)
-        if len(x.rstrip()) > 0:
-            tqdm.write(x, file=self._wrapped, nolock=nolock)
+        pre, sep, post = x.rpartition("\n")
+        if sep:
+            tqdm.write("".join(self._buf) + pre,
+                       file=self._wrapped, nolock=nolock)
```

Comment:
these lines can fail if e.g. `x` is `bytes`.

Reply:
fair enough, should be fixed now.